internal/app/networkd/pkg/address/kernel.go

```diff
@@ -0,0 +1,181 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package address
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/talos-systems/talos/internal/pkg/kernel"
+)
+
+// Kernel implements the Addressing interface
+type Kernel struct {
+	Addr              string
+	ServerAddress     string
+	Gateway           string
+	Netmask           string
+	FQDN              string
+	Device            string
+	Auto              string
+	PrimaryResolver   string
+	SecondaryResolver string
+	NTPServer         string
+}
+
+// Discover doesnt do anything in the static configuration since all
+// the necessary configuration data is supplied via config.
+// nolint: gocyclo
+func (k *Kernel) Discover(ctx context.Context) error {
+	var (
+		option *string
+		err    error
+	)
+
+	if option = kernel.ProcCmdline().Get("ip").First(); option == nil {
+		return fmt.Errorf("%s", "no kernel.ip argument supplied, skipping")
+	}
+
+	// TODO may need to move this earlier in networkd; if ip.dhcp is specified
+	// we shouldnt do anything because we dont know which (all?) interfaces the kernel
+	// will autoconfigure
+	if *option == "dhcp" {
+		return fmt.Errorf("unsupported kernel.ip configuration method: %s", *option)
+	}
+
+	// https://www.kernel.org/doc/Documentation/filesystems/nfs/nfsroot.txt
+	// ip=<client-ip>:<server-ip>:<gw-ip>:<netmask>:<hostname>:<device>:<autoconf>:<dns0-ip>:<dns1-ip>:<ntp0-ip>
+	fields := strings.Split(*option, ":")
+
+	if len(fields) < 4 {
+		return fmt.Errorf("invalid kernel option syntax: found %d fields but there must be at least 4", len(fields))
+	}
+
+	for idx, field := range fields {
+		switch idx {
+		case 0:
+			k.Addr = field
+		case 1:
+			k.ServerAddress = field
+		case 2:
+			k.Gateway = field
+		case 3:
+			k.Netmask = field
+		case 4:
+			k.FQDN = field
+		case 5:
+			k.Device = field
+		case 6:
+			k.Auto = field
+		case 7:
+			k.PrimaryResolver = field
+		case 8:
+			k.SecondaryResolver = field
+		case 9:
+			k.NTPServer = field
+		}
+	}
```

Comment:
If we pull this out as a separate function that returns `*option` (`option`) we can use `kerrnel.NewCmdline` in the tests instead of exposing `var Cmdline` at the package level. We don't have the linter on now, but public package level env vars would ding us because they are generally considered bad. See https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables for a good read on this.